packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts

```diff
@@ -870,10 +877,14 @@ export class UserPool extends UserPoolBase {
     }
 
     const userPoolId = arnParts.resourceName;
+    // ex) cognito-idp.us-east-1.amazonaws.com/us-east-1_abcdefghi
+    const providerName = `cognito-idp.${Stack.of(scope).region}.amazonaws.com/${userPoolId}`;
```

Comment:
The formation logic of this string should be changed, since `amazonaws.com` is a partition that could be different for different users. Can you use the `arnParts` attributes for this instead? Something along the lines of:
```
`cognito-idp.${arnParts.region}.${arnParts.partition}/${userPoolId}`;
```